features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/stored-procedures/mssql/action-cleanup/mssql-stale-action-cleanup.sql

```diff
@@ -0,0 +1,131 @@
+-- ------------------------------------------
+-- This PROCEDURE will cleanup the IDN_ACTION table by removing the entries which are not referenced by IDP_AUTHENTICATOR_PROPERTY table.
+-- Refer for more information: https://github.com/wso2-enterprise/asgardeo-product/issues/27966
+-- ------------------------------------------
+
+CREATE OR ALTER PROCEDURE WSO2_IDN_ACTION_CLEANUP AS
+BEGIN
+    -- ------------------------------------------
+    -- DECLARE VARIABLES
+    -- ------------------------------------------
+    DECLARE @batchSize INT
+    DECLARE @chunkSize INT
+    DECLARE @batchCount INT
+    DECLARE @chunkCount INT
+    DECLARE @rowCount INT
+    DECLARE @enableLog BIT
+    DECLARE @backupTables BIT
+
+    -- ------------------------------------------
+    -- CONFIGURABLE VARIABLES
+    -- ------------------------------------------
+    SET @batchSize = 10000 -- SET BATCH SIZE TO AVOID TABLE LOCKS [DEFAULT : 10000]
+    SET @chunkSize = 500000 -- CHUNK WISE DELETE FOR LARGE TABLES [DEFAULT : 500000]
+    SET @enableLog = 0 -- ENABLE LOGGING [DEFAULT : 0]
+    SET @backupTables = 0 -- SET IF IDN_ACTION TABLE NEEDS TO BE BACKED-UP BEFORE DELETE [DEFAULT : 0]
+
+    SET @rowCount = 0
+    SET @batchCount = 1
+    SET @chunkCount = 1
+
+    IF (@enableLog = 1)
+    BEGIN
+        SELECT '[' + convert(varchar, getdate(), 121) + '] WSO2_IDN_ACTION_CLEANUP() STARTED...!' AS 'INFO LOG'
```

Comment:
Since this only deletes the stale actions of AUTHENTICATION action type, shall we specifically mention that. Otherwise people may mislead that this will delete all the stale actions.
@malithie WDYT?